### ISSUE TITLE
Spreadsheet: Ensure session IDs are unique

### ISSUE THREAD

**Issue by mmmavis** [mozilla/mozfest-event-app#328]
In the event app spreadsheet, each Space will have its own sheet/tab. Sessions exported from the [mozfest-program-2016](https://github.com/MozillaFoundation/mozfest-program-2016/issues) repo will have issue number as their session ID. For session entries manually added after the export process we'll have to think of a way to ensure their IDs are **globally unique** (not just unique with a particular Space).

**Comment by mmmavis** [comment]
Idea #1 

Create a "ID tracker" tab that logs all session IDs that ever appear in the event app spreadsheet. Create a validator to check against "ID tracker" when creating new session entries.

**Comment by xmatthewx** [comment]
You might have the first two digits represent the space. e.g. 2003 = a session in connected devices. 3017 = a session in youth zone. Even if all the sessions are sequential, appending a space prefix would help prevent collisions.

**Comment by mmmavis** [comment]
I created a "ID Tracker" tab which lists all existing sessions IDs in the spreadsheet. I'm asking wranglers to visit that tab and make sure any **NEW** session ID they are entering isn't a duplicate.

**Comment by xmatthewx** [comment]
I recommend we make a plan and post it to a wiki page for everyone's easy reference. 

It should cover editing vs adding sessions, for both now vs later.

I also suggest that all new session requests happen via Github. All the wranglers know github. It's too hard to track things via chat, email, docs. 

cc @edrushka

**Comment by mmmavis** [comment]
@Saallen 

> I have one question on the guide- Session ID- should we give each space an alllocation
such as Delimmas new id no start at 3XX
Badges 15xxx
I worry about giving them the huge option of any number...

Existing sessions (sessions we ported from Github) have Github ticket numbers as their IDs. I'm worried if new sessions start their ID from 5xx, 10xx etc it would confuse people as IDs within each Space will look like a pool of random numbers.

Last year we asked people to go to the program repo, create a ticket for any new session they were adding, and grab the Github ticket number as the ID - not many people followed the instructions 😞 

I'm hoping using the "ID tracker" tab I created as the look up chart should reduce fewer human errors... I hope that would force people to do a lookup **every time** they add a new session. WE JUST NEED TO PLANT THIS INSTRUCTION IN THEIR HEAD (I'll highlight in more in the guide with note that ID can only be number) 😄 If this doesn't work out or it creates more confusion, I think we can ask them to leave ID blank and having me and/or someone to be the "ID assigner". I won't be fully available at MozFest so someone else has to be the go-to person if we fall back to this plan.

/cc @xmatthewx @edrushka

**Comment by xmatthewx** [comment]
I believe we agreed that we would lock the spreadsheet on Oct 7th. True? After that, we said there should be a single festival coordinator available to manage changes to the spreadsheet. I think wranglers should file issues. We can add an issue template to mention fest coordinator for triage.

Before now and Oct 7th, it might be easiest if people don't assign IDs and we do that for them.

**Comment by edrushka** [comment]
- We've extended the deadline. Spreadsheet will now be locked **EOD Oct 12.**
- Yes, we (the production team) intends to have a single festival coordinator in place to manage the changes.
- +1 to template that tags coordinator. 
- We already told the Space Wranglers to follow Mavis' instructions for assigning new IDs. It will likely be _more_ confusing to email them with different instructions now.

**Comment by mmmavis** [comment]
Hey @xmatthewx, sorry for the confusion. @Saallen and I chatted earlier today and we decided to extend the deadline till Oct 12. We wanted to give wranglers more time so that they can fill out location info their own. After Oct 12, we will disable wranglers' edit access and changes will have to done by the festival coordinator.

I created https://github.com/mozilla/mozfest-event-app/issues/371 and soon we'll be able to see real (though WIP) schedule on our staging app. From now till Oct 12 I can still work on the script (spreadsheet -> app) to make sure things get properly parsed.

Let's stick with the original ID plan so we can avoid confusing 40+ people. 😂
